### PR TITLE
Fix cypress inconsistent tests with longer wait and purecomponents

### PR DESCRIPTION
**Pull request by fuziontech**
This is a test to see if we rerun tests with this configuration, will they be less flakey?

### CODE REVIEW

**Comment by macobo** [comment]
Fly-by: It seems from the code that the issue might be with animations?

In that case, one thing I've done in the past is turn off all animations when feature specs are running. 

```
<% if Rails.env.test? %>
  <style>
    * {
      transition: none !important;
      transition-property: none !important;
      transition-duration: .0s !important;
      -webkit-animation-duration: .0s !important;
      animation-duration: .0s !important;
    }
  </style>
<% end %>
```

If the issue comes from a react component with timing, a similar trick could be used to put something on the window scope. For example, setting `window.env = 'test'`, wrap the animating component and ignore the delay prop when `window.env === 'test'`

File: cypress/integration/trends_elements.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ describe('Trends actions & events', () => {
     it('Save to dashboard', () => {
         cy.get('[data-attr=save-to-dashboard-button]').click()
         cy.contains('Add panel to dashboard').click()
-        cy.wait(500) // not ideal but toast has a delay render
+        cy.wait(700) // not ideal but toast has a delay render
         cy.get('[data-attr=success-toast]').should('exist')
```
Comment (macobo) [inline cypress/integration/trends_elements.js:104]: Should this not wait already? Is the wait above to avoid timing out?

**Comment by macobo** [comment]
If the issue is with ajax timing then another hack I've done in the past is this:

```typescript
import axios from 'axios'
import { getCsrfToken } from 'src/shared/csrfToken'

declare global {
  interface Window { _activeXhrQueries: number }
}

// Snip

// Keep track of active async requests (used in feature tests to wait for async queries)
window._activeXhrQueries = 0
axios.interceptors.request.use((config) => {
  window._activeXhrQueries++
  return config
}, (error) => {
  window._activeXhrQueries--
  return Promise.reject(error)
})
axios.interceptors.response.use((response) => {
  window._activeXhrQueries--
  return response
}, (error) => {
  window._activeXhrQueries--
  return Promise.reject(error)
})

export default axios
```

And in test code sleep until `window._activeXhrQueries` is zero. This might not work due to polling /stats all the time though.

However I'd hope cypress would help avoid this sort of ugliness.

File: cypress/integration/trends_elements.js (hunk 1 above)
Comment (EDsCODE) [reply cypress/integration/trends_elements.js:104]: cypress should wait when i originally wrote these tests it wasn't catching the toast without the wait

**Comment by EDsCODE** [comment]
@fuziontech should we run a few more times and get this merged?

**Pull request merged**